homework2/task1/main.c

```diff
@@ -0,0 +1,116 @@
+#include <stdio.h>
+#include <time.h>
+#include <stdbool.h>
+
+#define AMOUNT_OF_FIBONACCI_NUMBERS 40
+
+unsigned int recursiveFibonacci(unsigned int number)
+{
+    if (number == 0)
+    {
+        return 0;
+    }
+    if (number == 1 || number == 2)
+    {
+        return 1;
+    }
+    return recursiveFibonacci(number - 1) + recursiveFibonacci(number - 2);
+}
+
+unsigned int linearFibonacci(unsigned int number)
+{
+    int number1 = 1, number2 = 1, number3 = 2;
+    if (number == 0)
+    {
+        return 0;
+    }
+    if (number == 1 || number == 2)
+    {
+        return 1;
+    }
+    for (int i = 3; i < number; i++)
```

Comment:
warning: comparison of integers of different signs: 'int' and 'unsigned int' [-Wsign-compare]